lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -592,27 +615,51 @@ monitorStakePools tr gp nl DBLayer{..} =
 monitorMetadata
     :: Tracer IO StakePoolLog
     -> GenesisParameters
-    -> (   PoolId
-        -> StakePoolMetadataUrl
-        -> StakePoolMetadataHash
-        -> IO (Maybe StakePoolMetadata)
-       )
     -> DBLayer IO
     -> IO ()
-monitorMetadata tr gp fetchMetadata DBLayer{..} = forever $ do
-    refs <- atomically (unfetchedPoolMetadataRefs 100)
-    successes <- fmap catMaybes $ forM refs $ \(pid, url, hash) -> do
-        fetchMetadata pid url hash >>= \case
-            Nothing -> Nothing <$ do
-                atomically $ putFetchAttempt (url, hash)
-
-            Just meta -> Just hash <$ do
-                atomically $ putPoolMetadata hash meta
-
-    when (null refs || null successes) $ do
-        traceWith tr $ MsgFetchTakeBreak blockFrequency
-        threadDelay blockFrequency
+monitorMetadata tr gp DBLayer{..} = do
+    initSettings <- atomically readSettings
+    sref <- newIORef initSettings
+    forever $ do
+        prevSettings <- readIORef sref
+        newSettings <- atomically readSettings
+        writeIORef sref newSettings
+
+        -- delete old metadata table if need be
+        case (poolMetadataSource prevSettings, poolMetadataSource newSettings) of
+            (_, FetchNone) -> liftIO $ atomically $ delPoolMetadata
```

Comment:
Why the special case for `FetchNone` ?

Reply:
`(FetchNone, FetchNone)` doesn't trigger cleanup. If this was an old database, both prevSettings and newSettings will be `FetchNone`, but the database will still be populated.